src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/transformers/SharedMemoryAccessTransformer.kt

```diff
@@ -33,7 +33,12 @@ internal class SharedMemoryAccessTransformer(
     lateinit var analyzer: AnalyzerAdapter
 
     override fun visitFieldInsn(opcode: Int, owner: String, fieldName: String, desc: String) = adapter.run {
-        if (isCoroutineInternalClass(owner) || isCoroutineStateMachineClass(owner)) {
+        if (
+            isCoroutineInternalClass(owner) ||
+            isCoroutineStateMachineClass(owner) ||
+            // when initializing our own fields in constructor, we do not want to track that
```

Comment:
please add to the comment that it will result in bytecode `VerificationError`

Reply:
done